eq-author/src/App/QuestionnaireDesignPage/index.js

```diff
@@ -81,31 +86,48 @@ export class UnwrappedQuestionnaireDesignPage extends Component {
     } = this.props;
     const { entityName, entityId } = match.params;
 
-    let sectionId, position;
-    if (entityName === SECTION) {
-      sectionId = entityId;
-      position = 0;
-    } else if (entityName === PAGE) {
-      for (let i = 0; i < questionnaire.sections.length; ++i) {
-        const section = questionnaire.sections[i];
-        const page = find(section.pages, { id: entityId });
-        if (page) {
-          sectionId = section.id;
-          position = section.pages.indexOf(page) + 1;
-          break;
+    let sectionId, position, selectedPage, selectedSection, selectedFolder;
+
+    switch (entityName) {
+      case SECTION:
+        sectionId = entityId;
+        position = 0;
+        break;
+
+      case QUESTION_CONFIRMATION:
+      case PAGE:
+        for (const section of questionnaire.sections) {
+          for (const folder of section.folders) {
+            for (const page of folder.pages) {
+              const comparatorID =
+                entityName === QUESTION_CONFIRMATION
+                  ? page.confirmation.id
+                  : page.id;
+              if (comparatorID === entityId) {
+                selectedPage = page;
+                selectedSection = section;
+                selectedFolder = folder;
+                break;
+              }
+            }
+          }
         }
-      }
-    } else if (entityName === QUESTION_CONFIRMATION) {
-      for (let i = 0; i < questionnaire.sections.length; ++i) {
-        const section = questionnaire.sections[i];
-        const page = find(section.pages, { confirmation: { id: entityId } });
-        if (page) {
-          sectionId = section.id;
-          position = section.pages.indexOf(page) + 1;
-          break;
+
+        if (selectedPage) {
+          sectionId = selectedSection.id;
+          position = selectedSection.folders.indexOf(selectedFolder) + 1;
         }
-      }
+
+        break;
+
+      case FOLDER:
+        // TODO
+        break;
+
+      default:
+      // TODO
```

Comment:
This has been left in for when we do the front end work, is it? If so that's fine.

Reply:
Yeah, just a placeholder so whoever picks it up can see where the new logic needs to go.